frame/evm/precompile/contracts/encoder/src/lib.rs

```diff
@@ -47,29 +57,93 @@ where
 		_target_gas: Option<u64>,
 		_context: &Context,
 	) -> core::result::Result<PrecompileOutput, ExitError> {
-		if input.len() < SELECTOR {
+		if input.len() < ACTION_LEN {
 			return Err(ExitError::Other("input length less than 4 bytes".into()));
 		}
-		let selector = &input[0..SELECTOR];
-		let call: T::Call = match selector {
-			_ if selector == <from_substrate_issuing::Pallet<T>>::digest() => {
-				from_substrate_issuing::Call::<T>::asset_burn_event_handle(input.to_vec()).into()
+		let pallet_digest = &input[0..PALLET_DIG_LEN];
+		let method_digest = &input[PALLET_DIG_LEN..ACTION_LEN];
+		let output = match pallet_digest {
```

Comment:
An unnecessary match I think.

Reply:
Simply use if-else here would be more concise?

Reply:
maybe we will have other pallets using this encoder precompile? Not only s2s and ethereum pallets.